File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/dotnet-isolated/EventHub/MultiHttpTriggerKafkaOutput.cs
using System;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;
using Microsoft.Azure.Functions.Worker.Http;
using System.Net;

namespace EventHub
{
    public class MultiHttpTriggerKafkaOutput
    {
        // KafkaOutputBinding sample
        // This KafkaOutput binding will create a topic "topic" on the LocalBroker if it doesn't exists.
        // Call this function then the KafkaTrigger will be trigged.
        [Function("MultiHttpTriggerKafkaOutput")]
        
        public static MultipleOutputTypeForBatch Output(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequestData req,
            FunctionContext executionContext)
        {
            var log = executionContext.GetLogger("HttpFunction");
            log.LogInformation("C# HTTP trigger function processed a request.");

            string message = req.FunctionContext
                                .BindingContext
                                .BindingData["message"]
                                .ToString();

            string message1 = req.FunctionContext
                                .BindingContext
                                .BindingData["message1"]
                                .ToString();

            string message2 = req.FunctionContext
                                .BindingContext
                                .BindingData["message2"]
                                .ToString();


            string responseMessage = string.IsNullOrEmpty(message)
                ? "This HTTP triggered function executed successfully. Pass a message in the query string"
                : $"Message {message} {message1} {message2} sent to the broker. This HTTP triggered function executed successfully.";
            var response = req.CreateResponse(HttpStatusCode.OK);
            response.WriteString(responseMessage);

            string[] messages = new string[3];
            messages[0] = message;
            messages[1] = message1;
            messages[2] = message2;

            return new MultipleOutputTypeForBatch()
            {
                Kevents = messages,
                HttpResponse = response
            };
        }
    }

    public class MultipleOutputTypeForBatch
    {
        [KafkaOutput("EventHubBrokerList",
            "e2e-kafka-dotnet-isolated-multi-eventhub",
            Username = "$ConnectionString",
            Password = "%EventHubConnectionString%",
            Protocol = BrokerProtocol.SaslSsl,
            AuthenticationMode = BrokerAuthenticationMode.Plain
        )]        
        public string[] Kevents { get; set; }

        public HttpResponseData HttpResponse { get; set; }
    }
}

File: test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/FunctionApps/java/EventHub/src/main/java/com/function/MultiHttpTriggerKafkaOutput.java
package com.function;

import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;


import java.util.Optional;
import java.util.*;

public class MultiHttpTriggerKafkaOutput {
    @FunctionName("MultiHttpTriggerKafkaOutput")
    public HttpResponseMessage input(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            @KafkaOutput(name = "kafkaOutput",
                         topic = "e2e-kafka-java-multi-eventhub", 
                         brokerList="EventHubBrokerList",
                         username = "$ConnectionString",
                         password = "%EventHubConnectionString%",
                         authenticationMode = BrokerAuthenticationMode.PLAIN,
                         protocol = BrokerProtocol.SASLSSL)  OutputBinding<List<String>> output,
            final ExecutionContext context) {
        context.getLogger().info("Java HTTP trigger processed a request.");

        // Parse query parameter
        String message = request.getQueryParameters().get("message");
        String message1 = request.getQueryParameters().get("message1");
        String message2 = request.getQueryParameters().get("message2");

        context.getLogger().info("Message:" + message + " Message1: " + message1 + " Message2: " + message2);

        List<String> allMessages = new ArrayList<String>();
        allMessages.add(message);
        allMessages.add(message1);
        allMessages.add(message2);
        output.setValue(allMessages);
        return request.createResponseBuilder(HttpStatus.OK).body("Message Sent, " + message +" " + message1 + " " + message2).build();
    }
}